### PR TITLE
refactor: cleanup e2e replicaset utils

### PR DESCRIPTION
**Pull request by draveness**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:

Refactor and clean up e2e framework rs utils 

Related to #76206 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc andrewsykim

### CODE REVIEW

File: test/e2e/framework/replicaset/wait.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2017 The Kubernetes Authors.
+Copyright 2019 The Kubernetes Authors.
```
Comment (andrewsykim) [inline test/e2e/framework/replicaset/wait.go]: Can we undo this? Moving files shouldn't touch copy right dates IIRC :)

Comment (draveness) [reply test/e2e/framework/replicaset/wait.go]: Seems like a git issue, `wait.go` is actually a new file which is quite different from the original `rs_util.go`. 🤣

Comment (neolit123) [reply test/e2e/framework/replicaset/wait.go]: yes, k8s keeps the original date.

**Review by neolit123 — COMMENTED** [review]
/approve
/hold

File: test/e2e/framework/replicaset/wait.go (hunk 1 above)
Comment (andrewsykim) [reply test/e2e/framework/replicaset/wait.go]: It's a new file but I believe the date still needs to be preserved if git sees it as a move

Comment (draveness) [reply test/e2e/framework/replicaset/wait.go]: Updated the date in the latest commit.

Comment (neolit123) [reply test/e2e/framework/replicaset/wait.go]: please squash the commits into one and force push, so that we have a single commit.

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Review by timothysc — APPROVED** [review]
/approve

**Pull request merged**